tests/www/test_role_based_auth.py

```diff
@@ -0,0 +1,241 @@
+        self.tearDownClass()
+        with conf_vars(
+            {("api", "auth_backend"): "airflow.api.auth.backend.role_based_auth"}
+        ):
+            settings.configure_orm()
+            self.session = settings.Session
+            self.app = application.create_app(testing=True)
+            self.admin_role = self.app.appbuilder.sm.find_role("Admin")  # pylint: disable=no-member
+            admin_username = "das_admin"
+            admin_email = "das_admin@fab.org"
+            self.admin_user = self.app.appbuilder.sm.find_user(  # pylint: disable=no-member
+                username=admin_username, email=admin_email
+            )
+            if not self.admin_user:
+                self.admin_user = self.app.appbuilder.sm.add_user(  # pylint: disable=no-member
+                    admin_username,
+                    "admin",
+                    "user",
+                    admin_email,
+                    self.admin_role,
+                    "general",
+                )
+            self.admin_user.roles = [self.admin_role]
+
+            viewer_username = "das_viewer"
+            viewer_email = "das_viewer@fab.org"
+            self.viewer_role = self.app.appbuilder.sm.find_role("Viewer")  # pylint: disable=no-member
+            self.viewer_user = self.app.appbuilder.sm.find_user(  # pylint: disable=no-member
+                username=viewer_username, email=viewer_email
+            )
+            if not self.viewer_user:
+                self.viewer_user = self.app.appbuilder.sm.add_user(  # pylint: disable=no-member
+                    viewer_username,
+                    "viewer",
+                    "user",
+                    viewer_email,
+                    self.viewer_role,
+                    "general",
+                )
+            self.viewer_user.roles = [self.viewer_role]
+            pool.create_pool("not_swimming_pool", 16, "a description")
+
+    @classmethod
+    def tearDownClass(cls):
+        pools = [
+            "test_delete_pool_unauthorized",
+            "test_delete_pool_authorized",
+            "test_create_pool_unauthorized",
+            "test_create_pool_authorized",
+        ]
+        for pool_name in pools:
+            try:
+                pool.delete_pool(pool_name)
+            except PoolNotFound:
+                pass
+
+        dags = [
+            "test_delete_dag_unauthorized",
+            "test_delete_dag_authorized",
+            "test_pause_dag_unauthorized",
+            "test_delete_dag_authorized",
+            "test_pause_dag_unauthorized",
+            "test_pause_dag_authorized",
+            "test_trigger_dag_unauthorized",
+        ]
+
+        for dag in dags:
+            try:
+                delete_dag.delete_dag(dag)
+            except DagNotFound:
+                pass
+
+    def create_test_dag(self, dag_id):
+        dag = DagModel(dag_id=dag_id)
+        self.session.add(dag)
+        self.session.commit()
+
+    def use_admin_role(self):
```

Comment:
How is the end user to set this attribute?

Reply:
This isn't accessible to users. It just allows the tests to toggle between Admin access and Viewer access.